### ISSUE TITLE
updating python-twitch

### ISSUE THREAD

**Issue by ingwinlu** [anxdpanic/plugin.video.twitch#140]
hey, long time, no commits

I am currently doing a more or less complete overhaul to python-twitch. I will try to keep the old API around with deprecation warnings, but the new one should be a lot more sexy!

https://github.com/ingwinlu/python-twitch/pull/10
Feedback regarding the pending changes is welcome.

When I am finished I will squash -> merge and tag v1.0.0. Then we need to try and import it here ;)

Reasons for the overhaul: 
someone added a searchGames function, but due to the ugly codebase I never felt like making the pr to update here, a restructuring and cleanup should help and I hope this plugin can profit from it as well.

**Comment by MrSprigster** [comment]
Yeah, it's been a while :). Thanks for for your continuing work. I haven't had time to put any effort into this unfortunately but ill be here to assist with the 1.0 import :D

Cheers,

Sprig

**Comment by ingwinlu** [comment]
I released `0.5.0` with the old functionality and the added search games function.

`1.0.0` is not backwards compatible in any way. In fact I removed a lot of functionality that should rather be in the client application instead of the library (exception handling for example).
It would have simply been to much work to keep all of the old functionality around.

The new interface should be straightforward and I hope it motivates somebody to port this plugin over.

//EDIT: also introduced `six` as a dependency, but that should be available in xbmc as well.
